src/Microsoft.ServiceHub.Framework/DelegatingServiceJsonRpcDescriptor.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.IO.Pipelines;
+using StreamJsonRpc;
+
+namespace Microsoft.ServiceHub.Framework;
+
+/// <summary>
+/// A delegating RPC descriptor for services that support JSON-RPC. By default this descriptor will delegate operations to the wrapped descriptor.
+/// </summary>
+/// <remarks>
+/// This implementation will also ensure inner descriptor is updated with any changes to multiplexing stream options or exception strategy.
+/// </remarks>
+[DebuggerDisplay("{" + nameof(DebuggerDisplay) + ",nq}")]
+public class DelegatingServiceJsonRpcDescriptor : ServiceJsonRpcDescriptor
```

Comment:
Why is this `JsonRpc` rather than just `Rpc`? Is this _only_ compatible with the `ServiceJsonRpcDescriptor` for some reason?

We should target `ServiceRpcDescriptor` everywhere we can, so that if at some future time, we want to use an RPC protocol other than `JSON-RPC` courtesy of StreamJsonRpc, we'll be free to do so.

And if we *can* change this, let's update the xml doc as well to remove the "JSON-RPC" reference.

Reply:
It is really only compatible with ServiceJsonRpcDescriptor since those overrides that we are delegating are part of JsonRpcDescriptor only.  

ServiceRpcDescriptor doesn't really have methods that requires delegating so if we have a descriptor using another protocol, we wouldn't need this class. It may require its own delegating implementation though based on methods specific to its protocol